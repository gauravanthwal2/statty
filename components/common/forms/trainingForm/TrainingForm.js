import uuid from "react-uuid";
import React, { useEffect, useRef, useState } from "react";
import Select from "react-select";
import PrimaryButton from "../../buttons/primaryButton/PrimaryButton";
import {
  AddTrainingButtonText,
  UpdateTrainingButtonText,
  AddTrainingHeader,
  UpdateTrainingHeader,
  trainingInformation,
  PrimaryPOCInformation,
  SecondaryPOCInformation,
  Mentors,
  additionalDocs,
  trainingPageFormFields,
} from "../../../../utils/constants/trainingPageConstants";
import styles from "./TrainingForm.module.css";
import { useDispatch, useSelector } from "react-redux";
import { getAllUsers } from "../../../../services/usersServices";
import {
  addNewTraining,
  updateTrainingData,
} from "../../../../redux/addTrainings/addTrainingSlice";
import SecondaryButton from "../../buttons/secondaryButton/SecondaryButton";
import Head from "next/head";

const {
  trainingInformationHeading,
  trainingCodeLabel,
  trainingNameLabel,
  startDateLabel,
  endDateLabel,
  typeLabel,
  phaseLabel,
  statusLabel,
} = trainingInformation;

const {
  PPC_InformationHeading,
  PPC_OracleIDLabel,
  PPC_FirstNameLabel,
  PPC_LastNameLabel,
  PPC_EmailLabel,
} = PrimaryPOCInformation;

const {
  SPC_InformationHeading,
  SPC_oracleIDLabel,
  SPC_FirstNameLabel,
  SPC_LastNameLabel,
  SPC_EmailLabel,
} = SecondaryPOCInformation;
const { mentorsHeading, oracleIDLabel } = Mentors;
const { additionalDocsHeading, docsLabel } = additionalDocs;

export default function TrainingForm({
  isEditMode,
  isEditable,
  closeModal,
  selected,
}) {
  const [usersData, setUsersData] = useState([]);
  const [files, setFiles] = useState([]);
  const [form, setForm] = useState(trainingPageFormFields);
  //Typeahead Search state
  const [POCPrimaryOracleID, setPOCPrimaryOracleID] = useState(null);
  const [POCSecondaryOracleID, setPOCSecondaryOracleID] = useState(null);
  const [mentors, setMentors] = useState(null);

  const PPC_Ref = useRef(null);
  const SPC_Ref = useRef(null);
  const mentorsRef = useRef(null);

  const dispatch = useDispatch();
  const {
    trainingCode,
    trainingName,
    startDate,
    endDate,
    type,
    phase,
    status,
    PPC_OracleID,
    PPC_FirstName,
    PPC_LastName,
    PPC_Email,
    SPC_oracleID,
    SPC_FirstName,
    SPC_LastName,
    SPC_Email,
    oracleID,
    docs,
  } = form;

  // Setting Users TypeHead Values
  const userSelectData = usersData.map((item) => {
    const name = item.OracleId;
    return {
      value: name,
      label: name,
    };
  });

  const mentorsData = usersData.map((item) => {
    const name = `${item?.OracleId} - ${item?.FirstName}`;
    return {
      value: name,
      label: name,
    };
  });

  useEffect(() => {
    (async () => {
      const data = await getAllUsers();
      setUsersData(data);
    })();
  }, []);

  const convertToBase64 = (file) => {
    return new Promise((resolve, reject) => {
      const fileReader = new FileReader();
      fileReader.readAsDataURL(file);
      fileReader.onload = () => {
        resolve(fileReader.result);
      };
      fileReader.onerror = (error) => {
        reject(error);
      };
    });
  };
  const handleFiles = async (e) => {
    const file = e.target.files[0];
    const base64 = await convertToBase64(file);
    console.log("base64b", base64);
    setFiles((files) => [...files, e.target.files]);
  };

  const onHandleChange = ({ target: { name, value } }) => {
    setForm((prev) => ({ ...prev, [name]: value }));
  };

  console.log(PPC_Ref);

  const clearModal = () => {
    setForm(trainingPageFormFields);
    PPC_Ref.current?.Select?.clearValue();
    SPC_Ref.current?.Select?.clearValue();
    mentorsRef.current?.Select?.clearValue();
  };

  const setUsersDetailsByOracleId = (e, isSecondary) => {
    setPOCPrimaryOracleID(e.value);
    setPOCSecondaryOracleID(e.value);

    const selectedUser = usersData.filter(
      (item) => item.OracleId == e.value
    )[0];
    console.log("selectedUser", selectedUser);
    if (!isSecondary) {
      setForm((prev) => ({
        ...prev,
        PPC_OracleID: selectedUser?.OracleId,
        PPC_FirstName: selectedUser?.FirstName,
        PPC_LastName: selectedUser?.LastName,
        PPC_Email: selectedUser?.Email,
      }));
    } else {
      setForm((prev) => ({
        ...prev,
        SPC_oracleID: selectedUser?.OracleId,
        SPC_FirstName: selectedUser?.FirstName,
        SPC_LastName: selectedUser?.LastName,
        SPC_Email: selectedUser?.Email,
      }));
    }
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const data = {
      ...form,
      oracleID: mentors?.map((item) => item?.value.split("-")[0]).join(","),
      docs: files,
      id: uuid(),
    };
    dispatch(addNewTraining(data));
    clearModal();
    closeModal();
  };
  const handleUpdate = (e) => {
    e.preventDefault();
    const data = {
      ...form,
      oracleID: mentors?.map((item) => item?.value.split("-")[0]).join(","),
      docs: files,
      id: uuid(),
    };
    dispatch(updateTrainingData({ id: selected?.id, data }));
    clearModal();
    closeModal();
  };

  // Setting Pre Populated Values in the Inputbox
  useEffect(() => {
    if (isEditMode && selected) {
      setForm({
        trainingCode: selected?.trainingCode,
        trainingName: selected?.trainingName,
        startDate: selected?.startDate,
        phase: selected?.phase,
        endDate: selected?.endDate,
        status: selected?.status,
        type: selected?.type,
      });
    }
  }, [selected, isEditMode]);

  return (
    <div className="container mt-3">
      <Head>
        <title>Trainings</title>
      </Head>
      <h4 className="text-secondary text-center fw-semibold mb-3">
        {isEditMode ? UpdateTrainingHeader : AddTrainingHeader}
      </h4>
      <form
        className={styles.trainingForm}
        onSubmit={isEditMode ? handleUpdate : handleSubmit}
      >
        <div className={styles.formSection}>
          <div className="mx-3">
            <p className="text-secondary fw-bold">
              {trainingInformationHeading}
            </p>
            <div className="mb-3 mt-3">
              <label htmlFor="trainingCode">
                {trainingCodeLabel} <span className={styles.required}>*</span>
              </label>
              <input
                type="text"
                className="form-control"
                id="trainingCode"
                name="trainingCode"
                value={trainingCode}
                aria-label="trainingCode"
                onChange={onHandleChange}
                maxLength={10}
                required={true}
                readOnly={isEditMode ? true : false}
              />
            </div>
            <div className="mb-3">
              <label htmlFor="trainingName">
                {trainingNameLabel} <span className={styles.required}>*</span>
              </label>
              <input
                type="text"
                className="form-control"
                id="trainingName"
                name="trainingName"
                value={trainingName}
                aria-label="trainingName"
                onChange={onHandleChange}
                maxLength={30}
                required={true}
              />
            </div>
            <div className="mb-3">
              <label htmlFor="startDate">
                {startDateLabel} <span className={styles.required}>*</span>
              </label>
              <input
                type="date"
                className="form-control"
                id="startDate"
                name="startDate"
                aria-label="startDate"
                value={startDate}
                onChange={onHandleChange}
                required={true}
              />
            </div>
            <div className="mb-3">
              <label htmlFor="endDate">
                {endDateLabel} <span className={styles.required}>*</span>
              </label>
              <input
                type="date"
                className="form-control"
                id="endDate"
                name="endDate"
                aria-label="endDate"
                value={endDate}
                onChange={onHandleChange}
                required={true}
              />
            </div>
            {!isEditMode && (
              <div className="mb-3">
                <label htmlFor="type">
                  {typeLabel} <span className={styles.required}>*</span>
                </label>
                <select
                  id="type"
                  className="form-control"
                  required={true}
                  name="type"
                  aria-label="type"
                  value={type}
                  onChange={onHandleChange}
                >
                  {["Instructor led", "Self placed"].map((item, idx) => (
                    <option value={item} key={idx}>
                      {item}
                    </option>
                  ))}
                </select>
              </div>
            )}
            {isEditMode && (
              <>
                <div className="mb-3 mt-3 ">
                  <label htmlFor="phase">
                    {phaseLabel} <span className={styles.required}>*</span>
                  </label>
                  <select
                    id="phase"
                    name="phase"
                    data-testid="select"
                    className="form-control"
                    aria-label="phase"
                    required={true}
                    value={phase}
                    onChange={onHandleChange}
                    // readOnly={isEditable}
                  >
                    {[
                      "Upcoming",
                      "Sessions in progress",
                      "Capstone phase",
                      "Assessment phase",
                      "Finished",
                    ].map((item, idx) => (
                      <option
                        value={item}
                        key={idx}
                        data-testid="select-option"
                      >
                        {item}
                      </option>
                    ))}
                  </select>
                </div>
                <div className="mb-3 mt-3 ">
                  <label htmlFor="status">
                    {statusLabel} <span className={styles.required}>*</span>
                  </label>
                  <select
                    id="status"
                    name="status"
                    value={status}
                    onChange={onHandleChange}
                    className="form-control"
                    aria-label="status"
                    required={true}
                  >
                    <option value="" selected disabled={true} hidden>
                      status
                    </option>
                    {["Yet to start", "In progress", "Finished"].map(
                      (item, idx) => (
                        <option key={idx} value={item}>
                          {item}
                        </option>
                      )
                    )}
                  </select>
                </div>
              </>
            )}
          </div>
          <div className="mx-3">
            <p className="text-secondary fw-bold">{PPC_InformationHeading}</p>
            <div className="mb-3 mt-3">
              <label htmlFor="POCPrimaryOracleID">
                {PPC_OracleIDLabel} <span className={styles.required}>*</span>
              </label>
              <Select
                defaultValue={POCPrimaryOracleID}
                onChange={(e) => setUsersDetailsByOracleId(e, false)}
                options={userSelectData}
                id="POCPrimaryOracleID"
                name="POCPrimaryOracleID"
                ref={PPC_Ref}
                placeholder=""
                aria-label="POCPrimaryOracleID"
                theme={(theme) => ({
                  ...theme,
                  borderRadius: "5px",
                  colors: {
                    ...theme.colors,
                  },
                })}
                className=""
              />
            </div>
            <div className="mb-3">
              <label htmlFor="POCPrimaryFirstName">
                {PPC_FirstNameLabel} <span className={styles.required}>*</span>
              </label>
              <input
                type="text"
                className="form-control"
                id="POCPrimaryFirstName"
                name="POCPrimaryFirstName"
                aria-label="POCPrimaryFirstName"
                value={PPC_FirstName}
                onChange={onHandleChange}
                required={true}
                readOnly={isEditable}
              />
            </div>
            <div className="mb-3">
              <label htmlFor="POCPrimaryLastName">
                {PPC_LastNameLabel} <span className={styles.required}>*</span>
              </label>
              <input
                type="text"
                className="form-control"
                id="POCPrimaryLastName"
                name="POCPrimaryLastName"
                aria-label="POCPrimaryLastName"
                value={PPC_LastName}
                onChange={onHandleChange}
                required={true}
                readOnly={isEditable}
              />
            </div>
            <div className="mb-3">
              <label htmlFor="POCPrimaryEmail">
                {PPC_EmailLabel} <span className={styles.required}>*</span>
              </label>
              <input
                type="email"
                className="form-control"
                id="POCPrimaryEmail"
                name="POCPrimaryEmail"
                aria-label="POCPrimaryEmail"
                value={PPC_Email}
                onChange={onHandleChange}
                required={true}
                readOnly={isEditable}
              />
            </div>
          </div>
          <div>
            <div>
              <p className="text-secondary fw-bold">{SPC_InformationHeading}</p>
              <div className="mb-3 mt-3">
                <label htmlFor="POCSecondaryOracleID">
                  {SPC_oracleIDLabel} <span className={styles.required}>*</span>
                </label>
                <Select
                  defaultValue={POCSecondaryOracleID}
                  onChange={(e) => setUsersDetailsByOracleId(e, true)}
                  options={userSelectData}
                  id="POCSecondaryOracleID"
                  name="POCSecondaryOracleID"
                  aria-label="POCSecondaryOracleID"
                  ref={SPC_Ref}
                  placeholder=""
                  theme={(theme) => ({
                    ...theme,
                    borderRadius: "5px",
                    colors: {
                      ...theme.colors,
                    },
                  })}
                  className=""
                />
              </div>
              <div className="mb-3">
                <label htmlFor="POCSecondaryFirstName">
                  {SPC_FirstNameLabel}{" "}
                  <span className={styles.required}>*</span>
                </label>
                <input
                  type="text"
                  className="form-control"
                  id="POCSecondaryFirstName"
                  name="POCSecondaryFirstName"
                  aria-label="POCSecondaryFirstName"
                  value={SPC_FirstName}
                  onChange={onHandleChange}
                  required={true}
                  readOnly={isEditable}
                />
              </div>
              <div className="mb-3">
                <label htmlFor="POCSecondaryLastName">
                  {SPC_LastNameLabel} <span className={styles.required}>*</span>
                </label>
                <input
                  type="text"
                  className="form-control"
                  id="POCSecondaryLastName"
                  name="POCSecondaryLastName"
                  aria-label="POCSecondaryLastName"
                  value={SPC_LastName}
                  onChange={onHandleChange}
                  required={true}
                  readOnly={isEditable}
                />
              </div>
              <div className="mb-3">
                <label htmlFor="POCSecondaryEmail">
                  {SPC_EmailLabel} <span className={styles.required}>*</span>
                </label>
                <input
                  type="email"
                  className="form-control"
                  id="POCSecondaryEmail"
                  aria-label="POCSecondaryEmail"
                  name="POCSecondaryEmail"
                  value={SPC_Email}
                  onChange={onHandleChange}
                  required={true}
                  readOnly={isEditable}
                />
              </div>
            </div>
            <div>
              <p className="text-secondary fw-bold">{mentorsHeading}</p>
              <div className="mb-3">
                <label htmlFor="mentorsOracleID">
                  {oracleIDLabel} <span className={styles.required}>*</span>
                </label>
                <Select
                  defaultValue={mentors}
                  onChange={setMentors}
                  options={mentorsData}
                  formatOptionLabel={mentorsData}
                  id="mentorsOracleID"
                  aria-label="mentorsOracleID"
                  name="mentorsOracleID"
                  ref={mentorsRef}
                  theme={(theme) => ({
                    ...theme,
                    borderRadius: "5px",
                    colors: {
                      ...theme.colors,
                    },
                  })}
                  className=""
                  placeholder=""
                  isMulti
                />
              </div>
              <p className="text-secondary fw-bold">{additionalDocsHeading}</p>
              <div className="mb-3 mt-3">
                <label htmlFor="attachments">{docsLabel}</label>
                <input
                  type="file"
                  data-testid="fileInput"
                  className="form-control"
                  id="attachments"
                  aria-label="attachments"
                  name="attachments"
                  // value={docs}
                  onChange={handleFiles}
                  accept=".jpg, .jpeg, .pdf"
                  multiple={true}
                />
              </div>
              <div className="mb-5">
                {isEditMode ? (
                  <PrimaryButton
                    text={UpdateTrainingButtonText}
                    type="submit"
                  />
                ) : (
                  <SecondaryButton text={AddTrainingButtonText} type="submit" />
                )}
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  );
}
