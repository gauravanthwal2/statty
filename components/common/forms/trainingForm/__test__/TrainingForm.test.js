import userEvent from "@testing-library/user-event";
import {
  renderWithProvider,
  cleanup,
  screen,
  fireEvent,
} from "../../../../../utils/test-utils";
import TrainingForm from "../TrainingForm";

function renderProvider(component) {
  return renderWithProvider(component);
}

describe("training form renders", () => {
  afterEach(() => cleanup());
  test("testing labels in the form", () => {
    const { getByText } = renderWithProvider(
      <TrainingForm isEditMode={false} isEditable={true} />
    );
    expect(getByText(/Training Code */i)).toBeInTheDocument();
    expect(getByText(/Training Name */i)).toBeInTheDocument();
    screen.debug();
  });

  test("add training text shows up on add button click", () => {
    const { getByRole } = renderWithProvider(
      <TrainingForm isEditMode={false} isEditable={true} />
    );
    expect(getByRole("button")).toHaveTextContent(/Add New Training/i);
  });

  test("update training text shows up on update button click ", () => {
    const { getByRole } = renderWithProvider(
      <TrainingForm isEditMode={true} isEditable={true} />
    );
    expect(getByRole("button")).toHaveTextContent(/Update Training/i);
  });

  test("check values on onChange handler", async () => {
    const { getByLabelText } = renderProvider(
      <TrainingForm isEditMode={false} isEditable={true} />
    );
    const trainingCode = getByLabelText(/Training Code */i);
    expect(trainingCode).toHaveProperty("required", true);
    await userEvent.type(trainingCode, "React101");
    expect(trainingCode).toHaveValue("React101");
  });

  test("testing select boxes", async () => {
    const { getByTestId, getByRole, getAllByTestId } = renderProvider(
      <TrainingForm isEditMode={true} isEditable={true} />
    );

    const phase = [
      "Upcoming",
      "Sessions in progress",
      "Capstone phase",
      "Assessment phase",
      "Finished",
    ];

    const select = getByTestId("select");
    await userEvent.selectOptions(select, phase[0]);
    expect(getByRole("option", { name: phase[0] }).selected).toBeTruthy();
    expect(getByRole("option", { name: phase[0] })).toHaveTextContent(
      "Upcoming"
    );
    await userEvent.selectOptions(select, phase[1]);
    expect(getByRole("option", { name: phase[1] }).selected).toBeTruthy();
    expect(getByRole("option", { name: phase[1] })).toHaveTextContent(
      "Sessions in progress"
    );
    await userEvent.selectOptions(select, phase[2]);
    expect(getByRole("option", { name: phase[2] }).selected).toBeTruthy();
    expect(getAllByTestId("select-option").length).toBe(5);
    // await userEvent.deselectOptions(select, phase[2]); //for non multiple elements
    // expect(getByRole("option", { name: phase[2] }).selected).toBe(false);
  });

  test("upload multiple files", async () => {
    const handleFiles = jest.fn();
    const { getByTestId, getByLabelText } = renderProvider(
      <>
        <label htmlFor="attachments">Documents</label>
        <input
          type="file"
          id="attachments"
          data-testid="fileInput"
          onChange={handleFiles}
          accept=".jpg, .jpeg, .pdf"
          multiple={true}
        />
      </>
    );

    //Mock files for the test
    const files = [
      new File(["logo"], "logo.jpeg", { type: "image/jpeg" }),
      new File(["background"], "background.jpg", { type: "image/jpg" }),
      new File(["resume"], "resume.pdf", { type: "application/pdf" }),
      new File(["statty"], "statty.pptx.", {
        type: "application/vnd.openxmlformats-officedocument.presentationml.presentation",
      }),
    ];

    const inputField = getByTestId("fileInput");
    const input = getByLabelText(/Documents/i);
    // await userEvent.upload(input, files);
    fireEvent.change(inputField, { target: { files } });

    expect(handleFiles).toHaveBeenCalledTimes(1);

    expect(input.files).toHaveLength(4);
    expect(input.files[0]).toBe(files[0]);
    expect(input.files[1]).toBe(files[1]);

    //Checking file names
    expect(inputField.files[0].name).toBe("logo.jpeg");
    expect(inputField.files[1].name).toBe("background.jpg");
    expect(inputField.files[2].name).toBe("resume.pdf");

    //Checking file types
    expect(inputField.files[0].type).toBe("image/jpeg");
    expect(inputField.files[1].type).toBe("image/jpg");
    expect(inputField.files[2].type).toBe("application/pdf");
  });

  test("renders Training page unchanged", () => {
    const { container } = renderProvider(
      <TrainingForm isEditMode={true} isEditable={true} />
    );
    expect(container).toMatchSnapshot();
  });
});
