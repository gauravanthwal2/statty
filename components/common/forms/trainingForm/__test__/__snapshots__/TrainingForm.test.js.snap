// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`training form renders renders Training page unchanged 1`] = `
<div>
  <div
    class="container mt-3"
  >
    <h4
      class="text-secondary text-center fw-semibold mb-3"
    >
      Update Existing Training
    </h4>
    <form
      class="trainingForm"
    >
      <div
        class="formSection"
      >
        <div
          class="mx-3"
        >
          <p
            class="text-secondary fw-bold"
          >
            1.Training Information
          </p>
          <div
            class="mb-3 mt-3"
          >
            <label
              for="trainingCode"
            >
              Training Code
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <input
              aria-label="trainingCode"
              class="form-control"
              id="trainingCode"
              maxlength="10"
              name="trainingCode"
              readonly=""
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="trainingName"
            >
              Training Name
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <input
              aria-label="trainingName"
              class="form-control"
              id="trainingName"
              maxlength="30"
              name="trainingName"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="startDate"
            >
              Start Date
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <input
              aria-label="startDate"
              class="form-control"
              id="startDate"
              name="startDate"
              required=""
              type="date"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="endDate"
            >
              End Date
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <input
              aria-label="endDate"
              class="form-control"
              id="endDate"
              name="endDate"
              required=""
              type="date"
              value=""
            />
          </div>
          <div
            class="mb-3 mt-3 "
          >
            <label
              for="phase"
            >
              Phase
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <select
              aria-label="phase"
              class="form-control"
              data-testid="select"
              id="phase"
              name="phase"
              required=""
            >
              <option
                data-testid="select-option"
                value="Upcoming"
              >
                Upcoming
              </option>
              <option
                data-testid="select-option"
                value="Sessions in progress"
              >
                Sessions in progress
              </option>
              <option
                data-testid="select-option"
                value="Capstone phase"
              >
                Capstone phase
              </option>
              <option
                data-testid="select-option"
                value="Assessment phase"
              >
                Assessment phase
              </option>
              <option
                data-testid="select-option"
                value="Finished"
              >
                Finished
              </option>
            </select>
          </div>
          <div
            class="mb-3 mt-3 "
          >
            <label
              for="status"
            >
              Status
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <select
              aria-label="status"
              class="form-control"
              id="status"
              name="status"
              required=""
            >
              <option
                disabled=""
                hidden=""
                value=""
              >
                status
              </option>
              <option
                value="Yet to start"
              >
                Yet to start
              </option>
              <option
                value="In progress"
              >
                In progress
              </option>
              <option
                value="Finished"
              >
                Finished
              </option>
            </select>
          </div>
        </div>
        <div
          class="mx-3"
        >
          <p
            class="text-secondary fw-bold"
          >
            2.POC Information (Primary)
          </p>
          <div
            class="mb-3 mt-3"
          >
            <label
              for="POCPrimaryOracleID"
            >
              Oracle ID
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <div
              class=" css-b62m3t-container"
              id="POCPrimaryOracleID"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-17-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class=" css-1ko62if-control"
              >
                <div
                  class=" css-319lph-ValueContainer"
                >
                  <div
                    class=" css-14el2xx-placeholder"
                    id="react-select-17-placeholder"
                  />
                  <div
                    class=" css-6j8wv5-Input"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-describedby="react-select-17-placeholder"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-label="POCPrimaryOracleID"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      class=""
                      id="react-select-17-input"
                      role="combobox"
                      spellcheck="false"
                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <input
                name="POCPrimaryOracleID"
                type="hidden"
                value=""
              />
            </div>
          </div>
          <div
            class="mb-3"
          >
            <label
              for="POCPrimaryFirstName"
            >
              First Name
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <input
              aria-label="POCPrimaryFirstName"
              class="form-control"
              id="POCPrimaryFirstName"
              name="POCPrimaryFirstName"
              readonly=""
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="POCPrimaryLastName"
            >
              Last Name
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <input
              aria-label="POCPrimaryLastName"
              class="form-control"
              id="POCPrimaryLastName"
              name="POCPrimaryLastName"
              readonly=""
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              for="POCPrimaryEmail"
            >
              Email
               
              <span
                class="required"
              >
                *
              </span>
            </label>
            <input
              aria-label="POCPrimaryEmail"
              class="form-control"
              id="POCPrimaryEmail"
              name="POCPrimaryEmail"
              readonly=""
              required=""
              type="email"
              value=""
            />
          </div>
        </div>
        <div>
          <div>
            <p
              class="text-secondary fw-bold"
            >
              3.POC Information (Secondary)
            </p>
            <div
              class="mb-3 mt-3"
            >
              <label
                for="POCSecondaryOracleID"
              >
                Oracle ID
                 
                <span
                  class="required"
                >
                  *
                </span>
              </label>
              <div
                class=" css-b62m3t-container"
                id="POCSecondaryOracleID"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-18-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-1ko62if-control"
                >
                  <div
                    class=" css-319lph-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-18-placeholder"
                    />
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-18-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="POCSecondaryOracleID"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-18-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="POCSecondaryOracleID"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
            <div
              class="mb-3"
            >
              <label
                for="POCSecondaryFirstName"
              >
                First Name
                 
                <span
                  class="required"
                >
                  *
                </span>
              </label>
              <input
                aria-label="POCSecondaryFirstName"
                class="form-control"
                id="POCSecondaryFirstName"
                name="POCSecondaryFirstName"
                readonly=""
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                for="POCSecondaryLastName"
              >
                Last Name
                 
                <span
                  class="required"
                >
                  *
                </span>
              </label>
              <input
                aria-label="POCSecondaryLastName"
                class="form-control"
                id="POCSecondaryLastName"
                name="POCSecondaryLastName"
                readonly=""
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                for="POCSecondaryEmail"
              >
                Email
                 
                <span
                  class="required"
                >
                  *
                </span>
              </label>
              <input
                aria-label="POCSecondaryEmail"
                class="form-control"
                id="POCSecondaryEmail"
                name="POCSecondaryEmail"
                readonly=""
                required=""
                type="email"
                value=""
              />
            </div>
          </div>
          <div>
            <p
              class="text-secondary fw-bold"
            >
              4.Mentors
            </p>
            <div
              class="mb-3"
            >
              <label
                for="mentorsOracleID"
              >
                Oracle ID
                 
                <span
                  class="required"
                >
                  *
                </span>
              </label>
              <div
                class=" css-b62m3t-container"
                id="mentorsOracleID"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-19-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-1ko62if-control"
                >
                  <div
                    class=" css-319lph-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-19-placeholder"
                    />
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-19-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="mentorsOracleID"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-19-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div>
                  <input
                    name="mentorsOracleID"
                    type="hidden"
                  />
                </div>
              </div>
            </div>
            <p
              class="text-secondary fw-bold"
            >
              Additional Documents
            </p>
            <div
              class="mb-3 mt-3"
            >
              <label
                for="attachments"
              >
                Documents
              </label>
              <input
                accept=".jpg, .jpeg, .pdf"
                aria-label="attachments"
                class="form-control"
                data-testid="fileInput"
                id="attachments"
                multiple=""
                name="attachments"
                type="file"
              />
            </div>
            <div
              class="mb-5"
            >
              <button
                class="primaryButton"
                data-testid="primary"
                type="submit"
              >
                Update Training
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
